stage1/rootfs/diagexec/diagexec.c

```diff
@@ -180,19 +181,39 @@ static void load_env(const char *env)
 
 int main(int argc, char *argv[])
 {
-	const char *root, *cwd, *env, *exe;
-	exit_if(argc < 5,
-		"Usage: %s /path/to/root /work/directory /env/file /to/exec [args ...]", argv[0]);
+	const char *root, *cwd, *env, *uid_str, *gid_str, *exe;
+	char **args;
+	uid_t uid;
+	gid_t gid;
+	exit_if(argc < 7,
+		"Usage: %s /path/to/root /work/directory /env/file uid gid /to/exec [args ...]", argv[0]);
 
 	root = argv[1];
 	cwd = argv[2];
 	env = argv[3];
-	exe = argv[4];
-
-	load_env(env);
+	uid_str = argv[4];
+	uid = atoi(uid_str);
```

Comment:
I don't handle that. I'm not sure what we should do in that case...
See also https://github.com/appc/spec/issues/231